pkg/vpcmodel/synthesisOutput.go

```diff
@@ -59,25 +59,54 @@ func getSynthesisSpec(groupedLines []*groupedConnLine) *spec.Spec {
 	s := spec.Spec{}
 	connLines := []spec.SpecRequiredConnectionsElem{}
 	externals := spec.SpecExternals{}
+	// map from string(src+dst+conn) to int, if bidirectional or not. just one direction will be in this map
+	bidirectionalMap := make(map[string]bool)
 	sortGroupedLines(groupedLines)
 
+	initBidirectionalMap(groupedLines, bidirectionalMap)
+
 	for _, groupedLine := range groupedLines {
+		if groupedLine.CommonProperties.Conn.isEmpty() {
+			continue
+		}
 		srcName, srcType := handleNameAndType(groupedLine.Src, externals)
 		dstName, dstType := handleNameAndType(groupedLine.Dst, externals)
-		if groupedLine.CommonProperties.Conn.isEmpty() {
+		bidirectional, ok := bidirectionalMap[dstName+srcName+groupedLine.CommonProperties.Conn.allConn.String()]
+
+		if !ok {
+			// it means that it is bidirectional but the conn line will be appended to the list with the other direction.
 			continue
 		}
-		// For now, ignoring responsiveness of TCP connection
+
 		connLines = append(connLines, spec.SpecRequiredConnectionsElem{
 			Src:              spec.Resource{Name: srcName, Type: srcType},
 			Dst:              spec.Resource{Name: dstName, Type: dstType},
-			AllowedProtocols: sortProtocolList(spec.ProtocolList(connection.ToJSON(groupedLine.CommonProperties.Conn.allConn)))})
+			AllowedProtocols: sortProtocolList(spec.ProtocolList(connection.ToJSON(groupedLine.CommonProperties.Conn.allConn))),
+			Bidirectional:    bidirectional})
 	}
 	s.Externals = externals
 	s.RequiredConnections = connLines
 	return &s
 }
 
+func initBidirectionalMap(groupedLines []*groupedConnLine, bidirectionalMap map[string]bool) {
+	for _, groupedLine := range groupedLines {
+		_, ok := bidirectionalMap[groupedLine.Src.SynthesisResourceName()+
+			groupedLine.Dst.SynthesisResourceName()+
+			groupedLine.CommonProperties.Conn.allConn.String()]
```

Comment:
It's dangerous to create a map key by concatenating strings without separators.
`{src: "ab", dst: "cde"}` will have the same key as `{src: "abc", dst: "de"}`